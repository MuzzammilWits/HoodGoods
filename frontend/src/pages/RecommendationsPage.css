/* frontend/src/pages/RecommendationsPage.css */
main.recommendations-page-container { /* Changed from .recommendations-page-container */
  padding: 20px;
  max-width: 1200px;
  margin: 0 auto;
  background-color: var(--background-color, #fff); /* Example using a CSS variable or default */
  min-height: calc(100vh - var(--navbar-height, 60px) - var(--footer-height, 100px)); /* Adjust based on your navbar/footer */
  font-family: 'Hind', sans-serif; /* Added a base font for consistency */
}

.recommendations-page-header {
  text-align: center;
  margin-bottom: 30px;
  padding-bottom: 20px;
  border-bottom: 1px solid var(--border-color-light, #eee);
}

.recommendations-page-header h1 {
  font-size: 2.2em;
  color: var(--text-primary, #333);
  margin-bottom: 10px;
  font-family: 'Poppins', sans-serif; /* Consistent heading font */
}

.recommendations-page-header p {
  font-size: 1.1em;
  color: var(--text-secondary, #666);
  line-height: 1.6; /* Improved readability */
}

.recommendation-section {
  margin-bottom: 40px;
  background-color: var(--accent-background-subtle, #f8f9fa); /* Subtle background for sections */
  padding: 20px; /* Add padding within each recommendation section */
  border-radius: 8px; /* Rounded corners for sections */
  box-shadow: 0 2px 5px rgba(0,0,0,0.05); /* Soft shadow for sections */
}

.recommendation-section:last-child {
  margin-bottom: 0;
}