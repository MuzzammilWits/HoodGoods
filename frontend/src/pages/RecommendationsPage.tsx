// frontend/src/pages/RecommendationsPage.tsx
import React from 'react';
import BestSellersList from '../components/recommendations/BestSellersList';
import { useNavigate } from 'react-router-dom';
import './RecommendationsPage.css'; // Create this CSS file

const RecommendationsPage: React.FC = () => {
  const navigate = useNavigate();
  return (
    // Changed div to main for the primary content container of the page
    <main className="recommendations-page-container" style={{ paddingTop: '20px', paddingBottom: '20px' }}>
        <h1 className="main-titles">Discover Products You Might Like</h1>
        <p className="main-titles-sub">Based on current trends and popular items.</p>
        <section className="back-button-container">
        <button onClick={() => navigate('/products')} className="back-button">
          Back to Products
        </button>
        </section>


      <section className="recommendation-section">
        <BestSellersList limit={12} timeWindowDays={30} title="Top Selling Products This Month" />
      </section>

      <section className="recommendation-section" style={{ marginTop: '40px' }}>
        <BestSellersList limit={8} timeWindowDays={7} title="Trending This Week" />
      </section>
    </main>
  );
};

export default RecommendationsPage;