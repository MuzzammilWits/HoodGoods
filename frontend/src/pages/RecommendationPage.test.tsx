// src/pages/RecommendationsPage.test.tsx
import { render, screen } from '@testing-library/react';
import { describe, test, expect, vi, beforeEach } from 'vitest';
import { BrowserRouter } from 'react-router-dom';
import RecommendationsPage from './RecommendationsPage';

// Mock the BestSellersList component with TypeScript support
vi.mock('../components/recommendations/BestSellersList', () => ({
  default: ({ limit, timeWindowDays, title }: { 
    limit: number; 
    timeWindowDays: number; 
    title: string 
  }) => (
    <div data-testid="mock-best-sellers">
      <h2 data-testid="section-title">{title}</h2>
      <p data-testid="limit-info">Limit: {limit}</p>
      <p data-testid="time-window-info">Time Window: {timeWindowDays} days</p>
    </div>
        
  ),
}));

describe('RecommendationsPage', () => {
  beforeEach(() => {
    render(
      <BrowserRouter>
        <RecommendationsPage />
      </BrowserRouter>
    );
  });

  test('renders page with correct main header and description', () => {
    // Check main page header
    const mainHeader = screen.getByRole('heading', { level: 1 });
    expect(mainHeader).toHaveTextContent('Discover Products You Might Like');
    
    // Check description text
    expect(screen.getByText('Based on current trends and popular items.')).toBeInTheDocument();
  });

  test('renders two recommendation sections with correct structure', () => {
  const sections = screen.getAllByTestId('mock-best-sellers');
  expect(sections).toHaveLength(2);
  
  // Get all elements with the test IDs
  const sectionTitles = screen.getAllByTestId('section-title');
  const limitInfos = screen.getAllByTestId('limit-info');
  const timeWindows = screen.getAllByTestId('time-window-info');
  
  // Check each section has the expected elements
  sections.forEach((section, index) => {
    expect(section).toContainElement(sectionTitles[index]);
    expect(section).toContainElement(limitInfos[index]);
    expect(section).toContainElement(timeWindows[index]);
  });
});

  test('first section displays monthly best sellers with correct props', () => {
    const sectionTitles = screen.getAllByTestId('section-title');
    expect(sectionTitles[0]).toHaveTextContent('Top Selling Products This Month');
    
    const limitInfos = screen.getAllByTestId('limit-info');
    expect(limitInfos[0]).toHaveTextContent('Limit: 12');
    
    const timeWindows = screen.getAllByTestId('time-window-info');
    expect(timeWindows[0]).toHaveTextContent('Time Window: 30 days');
  });

  test('second section displays weekly trending with correct props', () => {
    const sectionTitles = screen.getAllByTestId('section-title');
    expect(sectionTitles[1]).toHaveTextContent('Trending This Week');
    
    const limitInfos = screen.getAllByTestId('limit-info');
    expect(limitInfos[1]).toHaveTextContent('Limit: 8');
    
    const timeWindows = screen.getAllByTestId('time-window-info');
    expect(timeWindows[1]).toHaveTextContent('Time Window: 7 days');
  });

  test('applies correct spacing between sections', () => {
    const sections = screen.getAllByTestId('mock-best-sellers');
    const secondSection = sections[1];
    
    // Check if the second section has additional margin top
    // Note: This might need adjustment based on your actual styling implementation
    expect(secondSection.parentElement).toHaveStyle('margin-top: 40px');
  });
    

  test('matches snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <RecommendationsPage />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});