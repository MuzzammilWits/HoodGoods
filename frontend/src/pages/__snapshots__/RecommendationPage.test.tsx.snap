// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RecommendationsPage > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="recommendations-page-container"
    style="padding-top: 20px; padding-bottom: 20px;"
  >
    <header
      class="recommendations-page-header"
    >
      <h1>
        Discover Products You Might Like
      </h1>
      <p>
        Based on current trends and popular items.
      </p>
    </header>
    <section
      class="recommendation-section"
    >
      <div
        data-testid="mock-best-sellers"
      >
        <h2
          data-testid="section-title"
        >
          Top Selling Products This Month
        </h2>
        <p
          data-testid="limit-info"
        >
          Limit: 12
        </p>
        <p
          data-testid="time-window-info"
        >
          Time Window: 30 days
        </p>
      </div>
    </section>
    <section
      class="recommendation-section"
      style="margin-top: 40px;"
    >
      <div
        data-testid="mock-best-sellers"
      >
        <h2
          data-testid="section-title"
        >
          Trending This Week
        </h2>
        <p
          data-testid="limit-info"
        >
          Limit: 8
        </p>
        <p
          data-testid="time-window-info"
        >
          Time Window: 7 days
        </p>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
